azerothcore/azerothcore-wotlk
# fix(Core/Player): fix check for existing corpse on respawn

## Changes Proposed:
- Re-add the existing corpse check on player respawn that was removed (probably by mistake?) here: https://github.com/azerothcore/azerothcore-wotlk/commit/f4c226423dc0811efccf8135b46bc7ff46aab940#diff-baadebd8cd1117ca48225f316a5ab3fd5fd55b20963394d302341147183db067L5238

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/8111
- Closes https://github.com/chromiecraft/chromiecraft/issues/1672

## SOURCE:


## Tests Performed:

- Applied C++ changes, built and tested on Windows 10 clients and server

## How to Test the Changes:

Basically the repro steps here https://github.com/azerothcore/azerothcore-wotlk/issues/8111

1. Queue for RDF with 5 characters, accept
2. Complete the dungeon
3. Leave the group with 4 characters
4. With the last character still in the instance, die, release spirit
5. Run back to the dungeon as a ghost
6. Die again, release spirit
7. Make sure you're a ghost and you can move

## Known Issues and TODO List:




## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
